File: lib/model/google_pay_configuration.dart
import 'package:json_annotation/json_annotation.dart';

part 'google_pay_configuration.g.dart';

/// Class responsible for storing information about GooglePay configuration
/// - [merchantId] - your merchant id in Tpay system
@JsonSerializable()
class GooglePayConfiguration {
  final String merchantId;

  GooglePayConfiguration({required this.merchantId});

  factory GooglePayConfiguration.fromJson(Map<String, dynamic> json) => _$GooglePayConfigurationFromJson(json);
  Map<String, dynamic> toJson() => _$GooglePayConfigurationToJson(this);
}

File: android/src/main/kotlin/com/tpay/model/googlePay/GooglePayConfiguration.kt
package com.tpay.model.googlePay

import org.json.JSONObject

data class GooglePayConfiguration(val merchantId: String) {
    companion object {
        private const val MERCHANT_ID = "merchantId"

        fun fromJson(json: JSONObject) = json.run {
            GooglePayConfiguration(merchantId = getString(MERCHANT_ID))
        }
    }
}